core/src/dird/job.cc

```diff
@@ -1843,30 +1843,30 @@ int CreateRestoreBootstrapFile(JobControlRecord* jcr)
   UaContext* ua;
   int files;
 
-  rx.bsr = new_bsr();
+  rx.bsr = std::make_unique<RestoreBootstrapRecord>();
   rx.JobIds = (char*)"";
   rx.bsr->JobId = jcr->previous_jr.JobId;
   ua = new_ua_context(jcr);
-  if (!CompleteBsr(ua, rx.bsr)) {
+  if (!CompleteBsr(ua, rx.bsr.get())) {
     files = -1;
     goto bail_out;
   }
-  rx.bsr->fi = new_findex();
-  rx.bsr->fi->findex = 1;
-  rx.bsr->fi->findex2 = jcr->previous_jr.JobFiles;
+  for (auto fi = 1; fi <= jcr->previous_jr.JobFiles; fi++) {
```

Comment:
```suggestion
  for (uint32_t fi = 1; fi <= jcr->previous_jr.JobFiles; fi++) {
```
Otherwise comparison of interegers with different signedness.